docs/resources/user_role_grant.md

```diff
@@ -0,0 +1,78 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "cockroach_user_role_grant Resource - terraform-provider-cockroach"
+subcategory: ""
+description: |-
+  A role grant for a user. This resource is recommended to be used when a user's roles are managed across multiple terraform projects or in conjunction with console UI granted roles. For authoritative management over a user's roles, use the userrolegrants user_role_grants resource.
+  As with all terraform resources, care must be taken to limit management of the same resource to a single project.
+---
+
+# cockroach_user_role_grant (Resource)
+
+A role grant for a user. This resource is recommended to be used when a user's roles are managed across multiple terraform projects or in conjunction with console UI granted roles. For authoritative management over a user's roles, use the [user_role_grants](user_role_grants) resource.
+
+ As with all terraform resources, care must be taken to limit management of the same resource to a single project.
+
+## Example Usage
+
+```terraform
+variable "user_id" {
+  type = string
+}
+
+resource "cockroach_user_role_grant" "admin_grant" {
+  user_id = var.user_id
+  role = {
+    role_name     = "CLUSTER_ADMIN",
+    resource_type = "CLUSTER",
+    resource_id   = cockroach_cluster.example.id
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `role` (Attributes) (see [below for nested schema](#nestedatt--role))
+- `user_id` (String) ID of the user to grant these roles to.
+
+<a id="nestedatt--role"></a>
+### Nested Schema for `role`
+
+Required:
+
+- `resource_type` (String) Type of resource. Allowed values are:
+  * ORGANIZATION
+  * CLUSTER
+  * FOLDER
+- `role_name` (String) Name of the role to grant. Allowed values are:
+  * DEVELOPER
+  * ADMIN
+  * BILLING_COORDINATOR
+  * ORG_ADMIN
+  * ORG_MEMBER
+  * CLUSTER_ADMIN
+  * CLUSTER_OPERATOR_WRITER
+  * CLUSTER_DEVELOPER
+  * CLUSTER_CREATOR
+  * FOLDER_ADMIN
+  * FOLDER_MOVER
+
+Optional:
+
+- `resource_id` (String) ID of the resource. Omit if resource_type is 'ORGANIZATION'.
```

Comment:
Required if resource is cluster/folder is probably useful info.

Reply:
yes, good point.  Let me know what you think of this:

     ID of the resource. Required if the resource_type is 'FOLDER' or 'CLUSTER'. It should be omitted otherwise.

Additionally, I believe we'll have some docs review before landing this.

Reply:
I like that wording